controllers/gce/controller/cluster_manager.go

```diff
@@ -212,17 +212,28 @@ func (c *ClusterManager) GC(lbNames []string, nodePorts []backends.ServicePort)
 }
 
 func getGCEClient(config io.Reader) *gce.GCECloud {
-	allConfig, err := ioutil.ReadAll(config)
-	if err != nil {
-		glog.Fatalf("Error while reading entire config: %v", err)
+	getConfigReader := func() io.Reader { return nil }
+
+	if config != nil {
+		allConfig, err := ioutil.ReadAll(config)
+		if err != nil {
+			glog.Fatalf("Error while reading entire config: %v", err)
+		}
+		glog.V(2).Infof("Using cloudprovider config file:\n%v ", string(allConfig))
+
+		getConfigReader = func() io.Reader {
+			return bytes.NewReader(allConfig)
+		}
+	} else {
+		glog.V(2).Infoln("Not using cloudprovider config file")
```

Comment:
optional nit: Update the message to say "No config file provided. Continuing with default values".
Config file not provided is different than not using config file.